formal-models/README.md

```diff
@@ -112,6 +112,215 @@ configuration:
+* `M` preparation messages (of type `PrepareRequest` or `PrepareResponse`) with
+  no `ChangeView3` denote that the majority of nodes have decided to commit which
+  denotes the safe transition to stage `II` can be performed and `Commit` message
+  can safely be sent even if there's `ChangeView1` message in the network.
+* `M` messages each of the type `PrepareRequest` or `PrepareResponse` or `ChangeView1`
+  where at least one message is of the type `ChangeView1` denote that at least `M`
+  nodes have reached the stage `I` and the node can safely take further steps.
+  The additional `| Commit | ≤ F ∪ | CV3 | > 0` condition requires the majority of
+  nodes not to have the `Commit` message to be sent so that it's still possible to
+  collect enough `ChangeView[2,3]` messages to change the view in further stages.
+  If so, then the safe transition to stage `II` can be performed and `ChangeView2`
+  message can safely be sent.
+
+#### Stage II
+
+Once the node has `Commit` or `ChangeView2` message sent, it enters the stage `II`
+of the consensus process and waits for at least `M` messages of stage `II`
+(`Commit` or `ChangeView2`) to perform the transition to the next stage. The set
+of accepted messages can be arranged in the following way:
+
+* `M` messages of `ChangeView2` type denote that `M` nodes have decided to change
+  their view directly after entering the stage `II` due to timeout while waiting
+  for the `Commit` messages which leads to immediate view changing.
+* `M` messages of type `Commit` denote that the majority of nodes have decided to
+  commit which denotes the block can be accepted immediately without entering the
+  stage `III`.
+* `M` messages each of the type `Commit` or `ChangeView2` where not more than `F`
+  messages are of the type `Commit` denotes that the majority of nodes decided to
+  change their view after entering the stage `II` and there's not enough `Commit`
+  messages to create the block (and produce the fork), thus, the safe transition
+  to stage `III` can be performed and `ChangeView3` message can safely be sent
+  even if there's `Commit` message in the network.
+
+In addition, the direct transition from `cv2` state to the `commitSent` state is
+added in case if it's clear that there's more than `F` nodes have decided to
+commit and no `ChangeView3` message has been received which means that it's possible
+to produce block in the current view.
+
+#### Stage III
+
+Unlike the basic dBFT 2.0 model where consensus node locks on the commit phase,
+stage `III` gives the ability to escape from the commit phase via collecting the set
+of `M` `ChangeView3` messages. This phase is reachable as soon as at least `M` nodes
+has reached the phase `II` (have `ChangeView2` or `Commit` messages sent) and if
+there's not enough `Commit` messages (<=`F`) to accept the block. This stage is
+added to avoid situation when the node is being locked on the commit whereas the
+majority (>`F`) is willing to go to the next view.
```

Comment:
It's not a majority!

Reply:
It's not, but I meant that block can't be collected with this set pf Commits anyway. Fixed.